docs/src/main/paradox/server-side/graceful-termination.md

```diff
@@ -0,0 +1,75 @@
+connections that are still being used will remain active.
+It only unbinds the port on which the http server has been listening. This allows applications to finish streaming any
+responses that might be still in flight and eventually terminate the entire system. If your application uses long-lived
+connections, this does mean that these can delay the termination of your system indefinitely.
+
+A better and more graceful solution to terminate an Akka HTTP server is to use the
+@java[@javadoc[ServerBinding.terminate(Duration)](akka.http.javadsl.ServerBinding#terminate-java.time.Duration-)]
+@scala[@scaladoc[ServerBinding.terminate(FiniteDuration)](akka.http.scaladsl.Http$$ServerBinding#terminate%28FiniteDuration%29:Future[HttpTerminated])]
+method, which not only performs the unbinding, but also
+handles replying to new incoming requests with (configurable) "terminating" HTTP responses.
+It also allows setting a deadline after which any connections that are still alive will be shut down forcefully.
+More precisely, termination works by following these steps:
+
+Step 1, the server port is unbound and no new connections will be accepted (same as invoking `unbind()`).
+Immediately the 
+@java[@javadoc[ServerBinding#whenTerminationSignalIssued](akka.http.javadsl.ServerBinding#whenTerminationSignalIssued--) `CompletionStage`]
+@scala[@scaladoc[ServerBinding#whenTerminationSignalIssued](akka.http.scaladsl.Http$$ServerBinding#whenTerminationSignalIssued:Future[Deadline]) `Future`]
+is completed.
+This can be used to signal parts of the application that the HTTP server is shutting down and they should clean up as well.
+Note also that for more advanced shut down scenarios you may want to use the @extref[Coordinated Shutdown](akka-docs:/actors.html#coordinated-shutdown) capabilities of Akka.
+
+Step 2, in flight requests will be handled. If a request is "in-flight" (being handled by user code), it is given `hardDeadline` time to complete.
+ 
+- if user code emits a response within the timeout, then this response is sent to the client
+- if it is a streaming response, it is also mandated that it shall complete within the deadline, and if it does not
+  the connection will be terminated regardless of status of the streaming response. This is because such response could be infinite,
+  which could trap the server in a situation where it could not terminate if it were to wait for a response to "finish".
+    - existing streaming responses must complete before the deadline as well.
+      When the deadline is reached the connection will be terminated regardless of status of the streaming responses.
+- if user code does not reply with a response within the deadline, we produce a special @java[`akka.http.javadsl.settings.ServerSettings.getTerminationDeadlineExceededResponse`]@scala[`akka.http.scaladsl.settings.ServerSettings.terminationDeadlineExceededResponse`]  HTTP response (e.g. `503 Service Unavailable`)
```

Comment:
superfluos comma and double space after class name

Reply:
thanks